cspatients/tests.py

```diff
@@ -356,3 +446,212 @@ def test_right_patient_id_bad_dict(self):
 
         patiententry = PatientEntry.objects.get(patient_id="XXXXX")
         self.assertEqual(patiententry, save_model_changes(changes_dict))
+
+
+class GetRPDictTest(TestCase):
+
+    """
+        Test the get_rp_dict method
+        Takes in request.POST method
+    """
+
+    def setUp(self):
+        self.client = Client()
+        self.post_query = parse.urlencode(SAMPLE_RP_POST_DATA, doseq=True)
+
+    def test_get_rp_event_dict(self):
+        """
+            The Method should be able to extract the name, and patient ID,
+            from the Query dict into a dictionary with just name and
+            patient_id
+        """
+        # Simulate posting the query string and getting a QueryDict
+        response_dict = QueryDict(query_string=self.post_query)
+
+        # Extract the dictionary using the get_rp_dict method
+        event_dict = get_rp_dict(response_dict)
+
+        # Test that it gets the appropriate keys and values
+
+        self.assertTrue(event_dict.__contains__("name"))
+        self.assertTrue(event_dict.__contains__("patient_id"))
+
+        # Test that it saves the right methods
+
+        self.assertTrue(event_dict["name"] == "Jane Doe")
+        self.assertTrue(event_dict["patient_id"] == "HLFSH")
+
+    def test_get_entrychanges_dict(self):
+        """
+            The Method when passed a context of entrychanges, must be able
+            to pick up the change category and the new value
+        """
+        # Simulate posting the query string and getting a QueryDict
+        response_dict = QueryDict(query_string=self.post_query)
+
+        # Extract the dictionary using the get_rp_dict method and
+        # entrychanges context
+        changes_dict = get_rp_dict(response_dict, context="entrychanges")
+
+        # Test that the change category and values are stored well
+        self.assertTrue(changes_dict.__contains__("name"))
+        self.assertTrue(changes_dict['name'] == "Nyasha")
+
+        # Test that the patient ID is kept well
+        self.assertTrue(changes_dict.__contains__("patient_id"))
+        self.assertTrue(changes_dict["patient_id"] == "HLFSH")
+
+
+# API Tests
+
+
+class RPEventTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.data = SAMPLE_RP_POST_DATA
+
+    def test_rpevent_saves_minimum_data_correctly(self):
+
+        delete_whole_table()
```

Comment:
Can you run me through the reasoning of running this function here?

Usually we would use the `setUp` function to make sure that the state of our application is in the desired format, and if necessary, we can add state/info/data on a per-test basis. So it seems counter-intuitive that we are removing data before we run our tests.

Reply:
I just read up that I do not need to manually reset the database as the TestCase class does that automatically. So I have removed the method